cfme/tests/optimize/test_bottlenecks.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+from datetime import timedelta
+
+from fixtures.pytest_store import store
+from cfme.optimize.bottlenecks import Bottlenecks
+from utils import conf
+from utils.appliance.implementations.ui import navigate_to
+from utils.appliance import get_or_create_current_appliance
+from utils.ssh import SSHClient
+from utils.version import current_version
+
+db_tbl = store.current_appliance.db['bottleneck_events']
+db_events = store.current_appliance.db.session.query(db_tbl.timestamp,
+    db_tbl.resource_type, db_tbl.resource_name, db_tbl.event_type, db_tbl.severity, db_tbl.message)
+
+
+@pytest.fixture(scope="session")
+def extend_db_partition():
+    app = get_or_create_current_appliance()
+    app.stop_evm_service()
+    app.extend_db_partition()
+    app.start_evm_service()
+
+
+@pytest.yield_fixture(scope="session")
+def backup_orig_state(extend_db_partition):
+    app = get_or_create_current_appliance()
+    app.backup_orig_state()
+    yield
+    app.restore_orig_state_from_backup()
+
+
+@pytest.fixture(scope="session")
+def db_restore(backup_orig_state):
+    app = get_or_create_current_appliance()
+    app.extend_db_partition()
+    app.stop_evm_service()
+    app.drop_database()
+    db_storage_hostname = conf.cfme_data['bottlenecks']['hostname']
+    db_storage = SSHClient(hostname=db_storage_hostname, **conf.credentials['bottlenecks'])
+    with db_storage as ssh:
+        # Different files for different versions
+        ver = "_56" if current_version() < '5.7' else ""
+        rand_filename = "/tmp/v2_key_{}".format(fauxfactory.gen_alphanumeric())
+        ssh.get_file("/home/backups/otsuman_db_bottlenecks/v2_key{}".format(ver), rand_filename)
+        dump_filename = "/tmp/db_dump_{}".format(fauxfactory.gen_alphanumeric())
+        ssh.get_file("/home/backups/otsuman_db_bottlenecks/db.backup{}".format(ver), dump_filename)
+        region_filename = "/tmp/REGION_{}".format(fauxfactory.gen_alphanumeric())
+        ssh.get_file("/home/backups/otsuman_db_bottlenecks/REGION{}".format(ver), region_filename)
+        guid_filename = "/tmp/GUID_{}".format(fauxfactory.gen_alphanumeric())
+        ssh.get_file("/home/backups/otsuman_db_bottlenecks/GUID{}".format(ver), guid_filename)
```

Comment:
`/home/backups/otsuman...` won't work but I believe that's my fault! :) I'll contact you on IRC with IP to upload these files to...